htmlproofer/plugin.py

```diff
@@ -59,8 +68,54 @@ def get_url_status(self, url: str, soup: BeautifulSoup) -> Tuple[str, int]:
             except requests.exceptions.ConnectionError:
                 return clean_url, -1
         else:
+            match = re.match(r'(.+)#(.+)', clean_url)
+            if match is not None:
+                # URL is a link to another local Markdown file that includes an anchor.
+                url_target, anchor = match.groups()
+                target_markdown = self.find_source_markdown(url_target, files)
+                if (target_markdown is not None
```

Comment:
so it's valid if the file is not found ?

Reply:
I was originally trying to be conservative so that if the algorithm failed to find the Markdown file it wouldn't fail the build, but this really should generate a 404. It likely will means there is a logic error in this plugin, because MkDocs already validates the relative links of Markdown *files*, just not the anchors, so I think MkDocs should fail before we'd ever legitimately get `None` for `target_markdown`.

Reply:
It's also possible that target file isn't in the `files` somehow. Anyway, adding a check for `None` won't bring in any false positive case, right ?

Reply:
@johnthagen in case you missed this

Reply:
@manuzhang Thank you for the reminder ping.

> It's also possible that target file isn't in the files somehow. Anyway, adding a check for None won't bring in any false positive case, right ?

As far as I can understand how MkDocs work, every Markdown that is properly part of the documentation project will be part of `files`. I tried this out on a fairly large project and this assumption held up.

I don't think checking for `None` should cause any false positive cases. If this fails for some reason it would mean that either the reference link is incorrect or there is a bug in this plugin in how it locates the files that we would need to fix.